tracing-core/tests/dispatch.rs

```diff
@@ -0,0 +1,118 @@
+use tracing_core::{dispatcher::*, metadata::Metadata, span, subscriber::Subscriber, Event};
+
+#[cfg(feature = "std")]
+#[test]
+fn set_default_dispatch() {
+    struct TestSubscriberA;
+    impl Subscriber for TestSubscriberA {
+        fn enabled(&self, _: &Metadata<'_>) -> bool {
+            true
+        }
+        fn new_span(&self, _: &span::Attributes<'_>) -> span::Id {
+            span::Id::from_u64(1)
+        }
+        fn record(&self, _: &span::Id, _: &span::Record<'_>) {}
+        fn record_follows_from(&self, _: &span::Id, _: &span::Id) {}
+        fn event(&self, _: &Event<'_>) {}
+        fn enter(&self, _: &span::Id) {}
+        fn exit(&self, _: &span::Id) {}
+    }
+    #[cfg(feature = "std")]
```

Comment:
AFAICT, the cfg attribute shouldn't be necessary here, as it's inside the test which already has that attribute.

 If we moved it out of the test functions so it can be used by both (as i suggested below), though, this _would_ be necessary.